core/src/main/java/hera/core/music/TrackScheduler.java

```diff
@@ -0,0 +1,114 @@
+package hera.core.music;
+
+import com.sedmelluq.discord.lavaplayer.player.AudioPlayer;
+import com.sedmelluq.discord.lavaplayer.player.event.AudioEventAdapter;
+import com.sedmelluq.discord.lavaplayer.tools.FriendlyException;
+import com.sedmelluq.discord.lavaplayer.track.AudioTrack;
+import com.sedmelluq.discord.lavaplayer.track.AudioTrackEndReason;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TrackScheduler extends AudioEventAdapter {
+
+	private List<AudioTrack> queue = new ArrayList<>();
+
+	private int queueIndex = 0;
+
+	private boolean loopQueue = false;
+
+	private long currentQueueMessageId = 0;
+
+	void queue(AudioPlayer player, AudioTrack track) {
+		queue.add(track);
+
+		// immediately play track if no other track is playing
+		if (player.getPlayingTrack() == null) {
+			queueIndex = queue.size() - 1;
+			player.playTrack(track);
+		}
+	}
+
+	public void skip(AudioPlayer player) {
+		// simulate the track ending
+		onTrackEnd(player, player.getPlayingTrack(), AudioTrackEndReason.FINISHED);
+	}
+
+	public void clearQueue() {
+		queueIndex = 0;
+		queue = new ArrayList<>();
+	}
+
+	@Override
+	public void onPlayerPause(AudioPlayer player) {
+		// Player was paused
+	}
+
+	@Override
+	public void onPlayerResume(AudioPlayer player) {
+		// Player was resumed
+	}
+
+	@Override
+	public void onTrackStart(AudioPlayer player, AudioTrack track) {
+		// A track started playing
```

Comment:
Should we announce the new song every time? (We did this with the old Hera, not sure tho if we still want this, as it is kinda spammy and you can just use `nowplaying` or `queue` if you want this info)